src/kyoto_reader/reader.py

```diff
@@ -134,8 +136,24 @@ def process_documents(self, doc_ids: List[str]) -> List[Optional[Document]]:
         parallel = Parallel(n_jobs=self.n_jobs)
         return parallel([delayed(KyotoReader._unwrap_self)(self, x) for x in doc_ids])
 
-    def process_all_documents(self) -> List[Document]:
-        return self.process_documents(self.doc_ids)
+    def process_documents_pool(self, doc_ids: List[str]) -> List[Optional[Document]]:
+        self_iter = repeat(self)
+        self_doc_ids_pair_iter = zip(self_iter, doc_ids)
+        with Pool() as pool:
+            return list(pool.starmap(KyotoReader._unwrap_self, self_doc_ids_pair_iter))
+
+    def process_all_documents(self, backend: Optional[str] = "joblib") -> List[Optional[Document]]:
```

Comment:
Since multiprocessing is faster than joblib, I think it is better to set the default backend "multiprocessing."